core/builder/src/main/java/io/quarkus/builder/ItemId.java

```diff
@@ -11,14 +11,22 @@
  */
 final class ItemId {
     private final Class<? extends BuildItem> itemType;
+    private final boolean multi;
+    private final boolean comparable;
 
     ItemId(final Class<? extends BuildItem> itemType) {
         Assert.checkNotNullParam("itemType", itemType);
         this.itemType = itemType;
+        this.multi = MultiBuildItem.class.isAssignableFrom(itemType);
+        this.comparable = Comparable.class.isAssignableFrom(itemType);
     }
 
     boolean isMulti() {
-        return MultiBuildItem.class.isAssignableFrom(itemType);
+        return multi;
+    }
+
+    boolean isComparable() {
+        return comparable;
```

Comment:
It doesn't seem worth wasting fields for what can be done in a run time type check.

Reply:
no problem.

Reply:
addressed in 351945e